src/checkers/cwe_248.ml

```diff
@@ -0,0 +1,89 @@
+    match already_found with
+      | Some(symb) -> Some(symb)
+      | None ->
+        match Jmp.kind instr with
+          | Goto _ | Ret _ | Int (_,_) -> None
+          | Call dst -> match Call.target dst with
+            | Direct tid ->
+              (* Tid.name tid is the name of the symbol.*)
+              Some(tid)
+            | _ -> None)
+
+(* Extract the Tid of a direct call, if the block contains a direct call. *)
+let extract_direct_call_symbol block =
+  match extract_direct_call_tid block with
+    | Some(tid) -> Some(Tid.name tid)
+    | None -> None
+
+(* check whether block contains a direct call to a symbol with name symbol_name *)
+let contains_symbol block symbol_name =
+  match extract_direct_call_symbol block with
+    | Some(symb) -> symb = symbol_name
+    | None -> false
+
+  (* Checks whether a subfunction contains a catch block. *)
+  let contains_catch subfunction =
+    let blocks = Term.enum blk_t subfunction in
+    Seq.exists blocks (fun block -> contains_symbol block "@__cxa_begin_catch")
+
+(* Find all calls to subfunctions that are reachable from this subfunction. The calls are returned
+    as a list, except for calls to "@__cxa_throw", which are logged as possibly uncaught exceptions. *)
+let find_calls_and_throws subfunction ~tid_map =
+  let blocks = Term.enum blk_t subfunction in
+  Seq.fold blocks ~init:[] ~f:(fun call_list block ->
+    if contains_symbol block "@__cxa_throw" then
+      let () = print_uncatched_exception  (Term.tid block) ~tid_map:tid_map in
+      call_list
+    else
+      match extract_direct_call_tid block with
+        | Some(tid) -> tid :: call_list
+        | None -> call_list
+  )
+
+(* find exception throws with for which an exception handler was not necessarily allocated beforehand.
+    The return value is a list of all already checked functions.*)
+let rec find_uncaught_exceptions subfunction already_checked_functions program ~tid_map =
+  if contains_catch subfunction then
+    (* This function contains a catch so we assume every throw reachable from here is catched. *)
+    already_checked_functions
+  else
+    let subfunction_calls = find_calls_and_throws subfunction ~tid_map:tid_map in
+    List.fold subfunction_calls ~init:already_checked_functions ~f:(fun already_checked subfunc ->
+      match List.exists ~f:(fun a -> a = subfunc) already_checked with
+        | true -> already_checked
+        | false ->  find_uncaught_exceptions ~tid_map:tid_map (Core_kernel.Option.value_exn (Term.find sub_t program subfunc)) (subfunc :: already_checked) program)
+
+
+let check_cwe program project tid_map symbol_pairs =
+  (* Get all subfunctions *)
+  let subfunctions = Term.enum sub_t program in
+  (* collect all entry points of the program *)
+  let entry_points = Seq.filter subfunctions ~f:(fun subfn -> Term.has_attr subfn Sub.entry_point) in
+  (* TODO: The _start entry point calls a libc-function which then calls the main function. Since right now only direct
```

Comment:
I think it would be better to move the TODOs out of the function, right above it. This would improve the readability.